src/mpi/init/mpir_init.c

```diff
@@ -228,6 +228,11 @@ int MPII_Init_thread(int *argc, char ***argv, int user_required, int *provided,
 
         if (device_count <= 0) {
             MPIR_CVAR_ENABLE_GPU = 0;
+        } else {
+            /* If the MPL backend doesn't support IPC, disable it for the upper layer */
+            if (!info.enable_ipc) {
+                MPIR_CVAR_CH4_IPC_GPU_P2P_THRESHOLD = -1;
```

Comment:
Note I changed the type for `ipc_attr.threshold.send_lmt_sz` to `int` to resolve the warning from this addition. This seemed like a reasonable solution since both threshold CVARs are also `int`.